pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -664,6 +686,29 @@ private RebalanceSummaryResult calculateDryRunSummary(Map<String, Map<String, St
     Set<String> serversGettingNewSegments = new HashSet<>();
     Map<String, RebalanceSummaryResult.ServerSegmentChangeInfo> serverSegmentChangeInfoMap = new HashMap<>();
     int segmentsNotMoved = 0;
+    Set<String> consumingSegments = new HashSet<>();
+    boolean isOfflineTable = TableNameBuilder.getTableTypeFromTableName(tableNameWithType) == TableType.OFFLINE;
+    Integer consumingSegmentsToBeMoved = null;
+    Integer maxBytesToCatchUpForConsumingSegments = null;
+    Map<String, Integer> bytesToCatchUpForSegments = null;
+    Map<String, Integer> bytesToCatchUpForServers = null;
+    if (!isOfflineTable) {
+      consumingSegmentsToBeMoved = 0;
+      bytesToCatchUpForSegments = getConsumingSegmentsBytesToCatchUp(tableNameWithType);
```

Comment:
this is an expensive operation and we don't really need to calculate this for REALTIME tables if consuming segments aren't to be moved. Can we only run this when we know for sure that CONSUMING segments are even involved in the rebalance? (i.e. make sure at least 1 consuming segment is moving)